crates/staged-sync/src/utils/init.rs

```diff
@@ -1,21 +1,34 @@
 use reth_db::{
     cursor::DbCursorRO,
     database::{Database, DatabaseGAT},
+    is_database_empty,
     mdbx::{Env, WriteMap},
     tables,
     transaction::{DbTx, DbTxMut},
+    version::{check_db_version_file, create_db_version_file, DatabaseVersionError},
 };
 use reth_primitives::{stage::StageId, Account, Bytecode, ChainSpec, H256, U256};
 use reth_provider::{
     AccountWriter, DatabaseProviderRW, PostState, ProviderFactory, TransactionError,
 };
-use std::{path::Path, sync::Arc};
+use std::{fs, path::Path, sync::Arc};
 use tracing::debug;
 
 /// Opens up an existing database or creates a new one at the specified path.
 pub fn init_db<P: AsRef<Path>>(path: P) -> eyre::Result<Env<WriteMap>> {
-    std::fs::create_dir_all(path.as_ref())?;
+    if is_database_empty(&path) {
+        fs::create_dir_all(&path)?;
```

Comment:
unrelated nit: we should add the file path as context here because io::Error does not